README.md

```diff
@@ -1,70 +1,213 @@
+```sh
+ sudo apt install git
+```
 
-### Advanced Configuration
+- Install nodejs and npm.
+
+```sh
+ sudo apt install nodejs
+```
+
+### Setup
+
+Clone this repository to your desired folder:
+
+```sh
+  git clone git@github.com:Kweeka1/metrics-app.git
+  cd metrics-app
+```
+
+### Install
+
+Install this project with:
+
+```sh
+  npm install
+```
+
+### Usage
+
+To run the project, follow below instructions:
+
+- Create an account at https://developer.riotgames.com/
+
+- Generate an api key
+
+- Create a new file in root directory called `.env` and insert below code:
+
+```sh
+REACT_APP_API_KEY={API_KEY_GENERATED}
+```
+
+- Example:
+```sh
+REACT_APP_API_KEY=RGAPI-2a9e6e8c-917c-8394-a8f3-baa489cd9fe8
+```
+
+- Now you can run the app in development mode by executing:
+
+```sh
+npm run start
+```
+
+- To get a production build, execute the following command
+
+```sh
+npm run build
+```
+
+### Run tests
+
+To run tests, run the following command:
+
+```sh
+  npm run test
+```
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/advanced-configuration](https://facebook.github.io/create-react-app/docs/advanced-configuration)
 
 ### Deployment
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/deployment](https://facebook.github.io/create-react-app/docs/deployment)
+You can deploy the production build of this project using npm library called `gh-pages`:
+
+- First install the library
+
+```sh
+npm install gh-pages --save-dev
+```
+
+- Then follow the instructions provided in their readme file to deploy the app
+
+[GH Pages README](https://www.npmjs.com/package/gh-pages)
+
+- deploy the build folder after running `npm run build`.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- AUTHORS -->
+
+## 👥 Authors <a name="authors"></a>
+
+👤 **Mohamed Talbi**
+
+- GitHub: [Mohamed Talbi](https://github.com/Kweeka1)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- FUTURE FEATURES -->
+
+## 🔭 Future Features <a name="future-features"></a>
+
+- [ ] **Improve User Interface**
+- [ ] **Make a desktop design for the app**
+- [ ] **Add expand button to match history card to show additional info**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- CONTRIBUTING -->
+
+## 🤝 Contributing <a name="contributing"></a>
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](https://github.com/Kweeka1/metrics-app/issues).
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- SUPPORT -->
+
+## ⭐️ Show your support <a name="support"></a>
+
+Give a ⭐️ if you like this project!
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to thank Microverse for project inspiration.
+
```

Comment:
- [ ] Kindly give appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).